rest_tools/utils/auth.py

```diff
@@ -67,22 +110,13 @@ def validate(self, token, **kwargs):
         Raises:
             Exception on failure to validate.
         """
-        try:
-            ret = jwt.decode(token, self.pub_secret, issuer=self.issuer, algorithms=[self.algorithm], **kwargs)
-        except jwt.exceptions.InvalidAudienceError:
-            if 'audience' not in kwargs:
-                kwargs['audience'] = ['ANY']
-                ret = jwt.decode(token, self.pub_secret, issuer=self.issuer, algorithms=[self.algorithm], **kwargs)
-            else:
-                raise
-        if 'type' not in ret:
-            raise Exception('no type information')
-        return ret
+        return self._validate(token, self.pub_secret, issuer=self.issuer, required=['type'], **kwargs)
```

Comment:
Is there any reason why you decided against calling (and creating) `super().validate(...)`?

Reply:
The main reason is that it's not meant to be used directly, so subclasses shouldn't be able to "pass" the method on to users.